zerver/lib/users.py

```diff
@@ -173,22 +172,9 @@ def is_administrator_role(role: int) -> bool:
     return role in {UserProfile.ROLE_REALM_ADMINISTRATOR, UserProfile.ROLE_REALM_OWNER}
 
 
-def bulk_get_users(
-    emails: List[str], realm: Optional[Realm], base_query: Optional[QuerySet[UserProfile]] = None
-) -> Dict[str, UserProfile]:
-    if base_query is None:
-        assert realm is not None
-        query = UserProfile.objects.filter(realm=realm, is_active=True)
-        realm_id = realm.id
-    else:
-        # WARNING: Currently, this code path only really supports one
-        # version of `base_query` being used (because otherwise,
-        # they'll share the cache, which can screw up the filtering).
-        # If you're using this flow, you'll need to re-do any filters
-        # in base_query in the code itself; base_query is just a perf
-        # optimization.
```

Comment:
For this second commit, correct, cross realm bot users are essentially immutable except with new versions changing the format of data structures.